packages/devtools_app/lib/src/timeline/timeline_flame_chart.dart

```diff
@@ -313,6 +329,60 @@ class FullTimelineFlameChartCanvas extends FlameChartCanvas<FullTimelineData> {
       currentRowIndex += sectionDepth;
       currentSectionIndex++;
     }
+
+    _calculateAsyncGuidelines();
+  }
+
+  void _calculateAsyncGuidelines() {
+    assert(rows.isNotEmpty);
+    assert(chartNodesByEvent.isNotEmpty);
+    verticalGuidelines.clear();
+    horizontalGuidelines.clear();
+    for (var row in rows) {
+      for (var node in row.nodes) {
+        if (node.data is AsyncTimelineEvent) {
+          final event = node.data as AsyncTimelineEvent;
+          if (event.hasOverlappingChildren) {
+            // Vertical guideline that will connect [node] with its overlapping
+            // children nodes. The line will end at [node]'s last child.
+            final verticalGuidelineX = node.rect.left + 1;
+            // Subtract [topOffset] so the coordinates of the guidelines are
+            // based on a starting y of 0.
+            final verticalGuidelineStartY = node.rect.bottom - topOffset;
+            final verticalGuidelineEndY =
+                chartNodesByEvent[event.children.last].rect.centerLeft.dy -
+                    topOffset;
+            verticalGuidelines.add(VerticalLineSegment(
+              Point(verticalGuidelineX, verticalGuidelineStartY),
+              Point(verticalGuidelineX, verticalGuidelineEndY),
+              'vertical guideline ${event.name} --> '
```

Comment:
remove these debugging strings or prefer to pass the event and event.child.last in to the VerticalLineSegment constructor as that would allow you to no just show a good tooltip but also navigate/select the relevant nodes.